### PR TITLE
Add document for kernel module requirements under platform setup

### PR DESCRIPTION
**Pull request by kfaseela**
Signed-off-by: Faseela K <faseela.k@est.tech>

Adding a prerequisite page for explicitly listing down the kernel modules that must be loaded on the cluster node running application pods with a Service Mesh proxy sidecar container. These modules are needed specifically by the istio-init container which sets up iptable rules in the pod to redirect any incoming or outgoing traffic towards the sidecar proxy in the istio-proxy container. 

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [X] Docs
- [X] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+## Kernel Module Requirements on Cluster Nodes
+
+The following kernel modules must be loaded on the cluster node running application pods with a Service Mesh proxy sidecar container.
+The modules are needed specifically by the `istio-init` container which sets up iptables rules in the pod to redirect any incoming or
+outgoing traffic towards the sidecar proxy in the istio-proxy container. While in many platforms, these seem to be automatically
+loaded, it is always good to make sure the prerequisites are met, as there were incidents reported where some of the specific
+modules listed down below were not loaded.
+
+| Module | Remark |
```
Comment (howardjohn) [inline content/en/docs/setup/platform-setup/prerequisites/index.md:23]: nit: there is probably some TPROXY ones required for TPROXY mode. Not sure what the are.

What did you use to find out the full set used?

**Review by howardjohn — COMMENTED** [review]
Content LGTM. Thanks!

File: content/en/docs/setup/platform-setup/prerequisites/index.md (hunk 1 above)
Comment (kfaseela) [reply content/en/docs/setup/platform-setup/prerequisites/index.md:25]: Finding the full set was a manual effort, used the golden files at tools/istio-iptables/pkg/capture/testdata/ for input and then with the descriptions about module loading from the iptables manpage gathered all those that would be loaded based on the reference files.

Eg:

[pkg/capture/testdata]$ grep -hoP -e '-t \S+' * | sort | uniq
-t mangle
-t nat
-t raw

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+keywords: [platform-setup,prerequisites]
+owner: istio/wg-environments-maintainers
+test: no
+---
+
+
+## Kernel Module Requirements on Cluster Nodes
+
+The following kernel modules must be loaded on the cluster node running application pods with a Service Mesh proxy sidecar container.
+The modules are needed specifically by the `istio-init` container which sets up iptables rules in the pod to redirect any incoming or
```
Comment (costinm) [inline content/en/docs/setup/platform-setup/prerequisites/index.md:18]: Or istio-cni deameon set. 

Also add that the modules are needed for iptables interception - Istio can also work in 'whitebox mode' where it doesn't intercept and doesn't need any special kernel module ( and in future may also use ebpf or other mechanisms ). Maybe a section describing the interception modes - and a table for each ?

I believe 'raw' is only used for DNS interception.

File: .spelling [hunk 3]
```diff
@@ -330,6 +330,7 @@ docker.io
 dogfood
 double-tls
 Drucker
+Dual-stack
```
Comment (ericvn) [inline .spelling:333]: Maybe lower case as I think we should lower case in the table (if not remove the `-` and then this entry isn't needed). I didn't see an upper case instance.

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+outgoing traffic towards the sidecar proxy in the istio-proxy container. While in many platforms, these seem to be automatically
+loaded, it is always good to make sure the prerequisites are met, as there were incidents reported where some of the specific
+modules listed down below were not available on the host or could not be automatically loaded by the iptables. For example,
+[`selinux issue`](https://www.suse.com/support/kb/doc/?id=000020241) talks about selinux in RHEL sometimes preventing
+the automatic loading of some of the below mentioned kernel modules.
+
+| Module | Remark |
+| --- | --- |
+| `br_netfilter` |  |
+| `ip6table_mangle` | Only needed for IPv6 or Dual-stack clusters |
```
Comment (ericvn) [inline content/en/docs/setup/platform-setup/prerequisites/index.md:28]: I'm not sure if dual stack or dual-stack is the correct way to write this. Either way, I think it should not be capitalized in the tables.

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+---
+title: Platform Prerequisites
+description: Prerequisites for platform setup for istio.
+weight: 1
+skip_seealso: true
+aliases:
+    - /docs/setup/kubernetes/prepare/platform-setup/prerequisites/
+    - /docs/setup/kubernetes/platform-setup/prerequisites/
```
[the remark this answers is not in the document]
Comment (kfaseela) [reply content/en/docs/setup/platform-setup/prerequisites/index.md]: Removed

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+keywords: [platform-setup,prerequisites]
+owner: istio/wg-environments-maintainers
+test: no
+---
+
+
```
[the remark this answers is not in the document]
Comment (kfaseela) [reply content/en/docs/setup/platform-setup/prerequisites/index.md:11]: it did not complain...should i change?

File: content/en/docs/setup/platform-setup/prerequisites/index.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+## Kernel Module Requirements on Cluster Nodes
+
+The following kernel modules must be loaded on the cluster node running application pods with a Service Mesh proxy sidecar container.
+The modules are needed specifically by the `istio-init` container which sets up iptables rules in the pod to redirect any incoming or
+outgoing traffic towards the sidecar proxy in the istio-proxy container. While in many platforms, these seem to be automatically
+loaded, it is always good to make sure the prerequisites are met, as there were incidents reported where some of the specific
+modules listed down below were not available on the host or could not be automatically loaded by the iptables. For example,
+[`selinux issue`](https://www.suse.com/support/kb/doc/?id=000020241) talks about selinux in RHEL sometimes preventing
```
[the remark this answers is not in the document]
Comment (kfaseela) [reply content/en/docs/setup/platform-setup/prerequisites/index.md]: Done

**Review by ericvn — APPROVED** [review]
LGTM

**Pull request merged**